apiserver/facades/client/secrets/secrets.go

```diff
@@ -331,3 +333,53 @@ func fromUpsertParams(p params.UpsertSecretArg) state.UpdateSecretParams {
 		ValueRef:    valueRef,
 	}
 }
+
+// CreateSecrets isn't on the v1 API.
+func (s *SecretsAPIV1) GrantSecret(_ struct{}) {}
+
+// GrantSecret grants access to a user secret.
+func (s *SecretsAPI) GrantSecret(arg params.GrantRevokeUserSecretArg) (params.ErrorResults, error) {
+	return s.secretsGrantRevoke(arg, s.secretsConsumer.GrantSecretAccess)
+}
+
+// RevokeSecret isn't on the v1 API.
+func (s *SecretsAPIV1) RevokeSecret(_ struct{}) {}
+
+// RevokeSecret revokes access to a user secret.
+func (s *SecretsAPI) RevokeSecret(arg params.GrantRevokeUserSecretArg) (params.ErrorResults, error) {
+	return s.secretsGrantRevoke(arg, s.secretsConsumer.RevokeSecretAccess)
+}
+
+type grantRevokeFunc func(*coresecrets.URI, state.SecretAccessParams) error
+
+func (s *SecretsAPI) secretsGrantRevoke(arg params.GrantRevokeUserSecretArg, op grantRevokeFunc) (params.ErrorResults, error) {
+	results := params.ErrorResults{
+		Results: make([]params.ErrorResult, len(arg.Applications)),
+	}
+
+	if err := s.checkCanAdmin(); err != nil {
```

Comment:
I think write access is perhaps all that's needed here. You need write access to deploy a charm. You should also be able to configure that charm as needed to make it work.

Reply:
But to grant access to the user(model level) secret, Admin access might be better?